Creiwry/kitten_project
# Add stripe checkout functionality

### Ticket Link
https://trello.com/c/gH3RF4aR
https://trello.com/c/yIMz78bQ

### User Story
https://trello.com/c/QxsqhYFh

### What does this PR do?
It:
- configures the stripe API
- adds routes for cart_items so that they are dependent on items as well
- creates backend logic for stripe checkout and order creation
- creates basic frontend elements for a user to get from root path to the stripe checkout button
- when stripe checkout is successful, it creates an order and empties user's cart
- add tests basic tests for the above functionalities in rspec

Additional:
It removes the assets/javascripts directory, as the application.js we are using in the project is in app/javascript. To ensure functionality, run

```bash
rails assets:precompile
```

Then, any javascript can be added and required in app/javascript/application.js 

### Where should the reviewer start?
The reviewer should start with the tests for the checkout controller and the cart model.
Most code was added in checkout_controller and in views.

### How should this be manually tested?
First, the reviewer should add the env variables PUBLISHABLE_KEY and SECRET_KEY to their .env
On the site, log in as a user, add one or more items to cart, and from cart check out to test if stripe works.

### Any background context you want to provide?
In the CartItem controller, the method create was updated. At the moment, the quantity attribute is hard-set to 1 in the controller method. This will need to be resolved later.
